### PR TITLE
[SYCLomatic] Add e2e test case for application llama.cpp migration

### PR DESCRIPTION
**Pull request by intwanghao**

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: .github/workflows/ci_win.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ jobs:
           - test_suite: samples
           - test_suite: api_coverage
           - test_suite: behavior_tests
+          - test_suite: applications
```
Comment (ShengchenJ) [inline .github/workflows/ci_win.yml]: We can add Windows applications test suite until there is a runnable Windows application.

File: applications/test_applications.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+def build_test():
+    if (os.path.exists(test_config.current_test)):
+        os.chdir(test_config.current_test)
+    srcs = []
+    cmp_opts = []
+    link_opts = []
+    objects = []
+
+    llama_files = ['common.cpp', 'sampling.cpp.dp.cpp', 'console.cpp', 'grammar-parser.cpp', 'train.cpp.dp.cpp',
```
Comment (ShengchenJ) [inline applications/test_applications.py:59]: How about use the --gen-build-script to generate the Makefile.dpct to build the migrated code?

File: applications/test_drivers.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testDrivers>
```
Comment (ShengchenJ) [inline applications/test_drivers.xml:2]: the file should be test_applications.py

**Review by ShengchenJ — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_win.yml (hunk 1 above)
Comment (intwanghao) [reply .github/workflows/ci_win.yml:32]: I think we can reserve the change here.

Comment (intwanghao) [reply .github/workflows/ci_win.yml:32]: Avoid others forgetting add test suit here if new case added.

File: applications/test_applications.py (hunk 2 above)
Comment (intwanghao) [reply applications/test_applications.py:59]: The --gen-build-script is not work well for llama.cpp, use --migrate-cmake-script is in progress.

File: .github/workflows/ci_win.yml (hunk 1 above)
Comment (intwanghao) [reply .github/workflows/ci_win.yml]: Done

File: applications/test_drivers.xml (hunk 3 above)
Comment (intwanghao) [reply applications/test_drivers.xml:2]: Done

**Pull request merged**